tests/unit/utils/test_pycrypto.py

```diff
@@ -1,103 +1,162 @@
 # -*- coding: utf-8 -*-
 
-# Import python libs
 from __future__ import absolute_import, print_function, unicode_literals
 
-import logging
 import re
 
-# Import Salt Libs
 import salt.utils.platform
 import salt.utils.pycrypto
 from salt.exceptions import SaltInvocationError
-
-# Import Salt Testing Libs
+from tests.support.mock import patch
 from tests.support.unit import TestCase, skipIf
 
-log = logging.getLogger(__name__)
-
 
 class PycryptoTestCase(TestCase):
     """
     TestCase for salt.utils.pycrypto module
     """
 
-    @skipIf(
-        not salt.utils.pycrypto.HAS_CRYPT or not salt.utils.pycrypto.HAS_PASSLIB,
-        "crypt not available",
-    )
-    def test_gen_hash(self):
+    passwd = "test_password"
+    expecteds = {
+        "sha512": {
+            "hashed": "$6$rounds=656000$goodsalt$25xEV0IAcghzQbu8TF5KdDMYk3b4u9nR/38xYU/26xvPgirDavreGhtLfYRYW.RngLmRtD9i8S8XP3dPx4.PV.",
+            "salt_crypt": "rounds=656000$goodsalt",
```

Comment:
I modified _gen_hash_passlib to work with rounds:

https://github.com/dhiltonp/salt/tree/pycrypto_gen_hash_rewrite